internal/cache/async.go

```diff
@@ -103,49 +103,41 @@ func (ac *asyncClient) doUpdate(ctx context.Context, key store.Key) {
 	switch {
 	case err == nil:
 		ac.objectStore.OverrideResourceVersionIfNewer(result)
-	case isRetryableError(err):
-		svc1log.FromContext(ctx).Warn("got retryable error, will retry", svc1log.Stacktrace(err))
-		ac.queue.AddIfAbsent(store.UpdateRequest(obj))
 	case errors.IsConflict(err):
 		svc1log.FromContext(ctx).Warn("got conflict, will try updating resource version", svc1log.Stacktrace(err))
-		newObj, getErr := ac.client.Get(key.Namespace, key.Name)
+		newObj, getErr := ac.client.Get(r.Key.Namespace, r.Key.Name)
 		switch {
 		case getErr == nil:
 			ac.objectStore.OverrideResourceVersionIfNewer(newObj)
-			ac.doUpdate(ctx, key)
-		case isRetryableError(getErr):
-			svc1log.FromContext(ctx).Warn("got retryable error, will retry", svc1log.Stacktrace(getErr))
-			ac.queue.AddIfAbsent(store.UpdateRequest(obj))
+			ac.doUpdate(ctx, store.UpdateRequest(newObj))
```

Comment:
Note that this resets the retry count, we can change that if this is not the behaviour we want.